pkg/controller/nebulascheduledbackup/nebula_scheduled_backup_manager.go

```diff
@@ -0,0 +1,396 @@
+	}
+
+	// Don't do anything except update backup cleanup scripts config map if cronjob is paused
+	if scheduledBackup.Spec.Pause != nil && *scheduledBackup.Spec.Pause {
+		klog.Infof("Nebula scheduled backup %s/%s is paused.", scheduledBackup.Namespace, scheduledBackup.Name)
+		return &kube.ScheduledBackupUpdateStatus{
+			CurrPauseStatus: scheduledBackup.Spec.Pause,
+		}, nil, nil
+	}
+
+	// Check status of most recent job
+	klog.Info("Checking status of most recent job")
+	var lastSuccessfulTime *metav1.Time
+	var lastBackupFailed bool
+
+	if len(successfulBackups) != 0 && len(failedBackups) != 0 {
+		sort.Sort(byBackupStartTime(successfulBackups))
+		sort.Sort(byBackupStartTime(failedBackups))
+
+		lastSuccessfulBackup := successfulBackups[len(successfulBackups)-1]
+		lastFailedBackup := failedBackups[len(failedBackups)-1]
+		if lastSuccessfulBackup.Status.TimeStarted.After(lastFailedBackup.Status.TimeStarted.Local()) {
+			// Most recent backup job succeeded
+			lastSuccessfulTime = lastSuccessfulBackup.Status.TimeCompleted
+			klog.Infof("Most recent backup job %s/%s for Nebula scheduled backup %s/%s succeeded", lastSuccessfulBackup.Namespace, lastSuccessfulBackup.Name, scheduledBackup.Namespace, scheduledBackup.Name)
+		} else {
+			// Most recent backup job failed
+			lastBackupFailed = true
+			klog.Errorf("Most recent backup job %s/%s for Nebula scheduled backup %s/%s failed", lastFailedBackup.Namespace, lastFailedBackup.Name, scheduledBackup.Namespace, scheduledBackup.Name)
+		}
+	} else if len(successfulBackups) != 0 {
+		lastSuccessfulBackup := successfulBackups[len(successfulBackups)-1]
+		lastSuccessfulTime = lastSuccessfulBackup.Status.TimeCompleted
+		klog.Infof("Most recent backup job %s/%s for Nebula scheduled backup %s/%s succeeded", lastSuccessfulBackup.Namespace, lastSuccessfulBackup.Name, scheduledBackup.Namespace, scheduledBackup.Name)
+	} else if len(failedBackups) != 0 {
+		lastFailedBackup := failedBackups[len(failedBackups)-1]
+		lastBackupFailed = true
+		klog.Errorf("Most recent backup job %s/%s for Nebula scheduled backup %s/%s failed", lastFailedBackup.Namespace, lastFailedBackup.Name, scheduledBackup.Namespace, scheduledBackup.Name)
+	} else {
+		klog.Info("No previous successful or failed backup jobs detected.")
+	}
+
+	// Check Nebula Graph cluster status and return error if it's down.
+	ns := scheduledBackup.GetNamespace()
+	if scheduledBackup.Spec.BackupTemplate.BR.ClusterNamespace != nil {
+		ns = *scheduledBackup.Spec.BackupTemplate.BR.ClusterNamespace
+	}
+
+	nc, err := bm.clientSet.NebulaCluster().GetNebulaCluster(ns, scheduledBackup.Spec.BackupTemplate.BR.ClusterName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get nebula cluster %s/%s err: %w", ns, scheduledBackup.Spec.BackupTemplate.BR.ClusterName, err)
+	}
+
+	if !nc.IsReady() {
+		return nil, nil, fmt.Errorf("nebula cluster %s/%s is not ready", ns, scheduledBackup.Spec.BackupTemplate.BR.ClusterName)
+	}
+
```

Comment:
nebula cluster status check can be after Pause

Reply:
Fixed